miio/viomidishwasher.py

```diff
@@ -0,0 +1,529 @@
+                    "freshdry_interval": 0,
+                    "wash_process": 0
+                }
+
+        """
+
+        self.data = data
+
+    @property
+    def child_lock(self) -> bool:
+        """Returns the child lock status of the device.
+
+        Returns:
+            bool: The return value. True for enabled, False for disabled.
+
+        Raises:
+            DeviceException: When the returned value by the device is not 1 or 0.
+        """
+        value = self.data["child_lock"]
+        if value in [0, 1]:
+            return bool(value)
+
+        raise DeviceException(f"{value} is not a valid child lock status.")
+
+    @property
+    def program(self) -> Program:
+        """Returns the current selected program of the device.
+
+        Returns:
+            Program: The current program or Program.Unknown if an Unknown program was detected.
+        """
+        program = self.data["program"]
+        try:
+            return Program(program)
+        except ValueError:
+            _LOGGER.warning("Program %r is Unknown.", program)
+            return Program.Unknown
+
+    @property
+    def door_open(self) -> bool:
+        """Returns the door status of the device.
+
+        Returns:
+            Bool: True if open False if closed.
+        """
+
+        return bool(self.data["run_status"] & (1 << 7))
+
+    @property
+    def system_status_raw(self) -> int:
+        """Returns the raw status number of the device.
+
+        This is in general used to detected:
+            - Errors in the system.
+            - If the door is open or not.
+
+        Returns:
+            int: Raw value.
+        """
+
+        return self.data["run_status"]
+
+    @property
+    def status(self) -> MachineStatus:
+        """Returns the machine status of the device.
+
+        This is in general used to determine what the device is doing.
+
+        Returns:
+            MachineStatus: The machine status Enum.
+        """
+
+        return MachineStatus(self.data["wash_status"])
+
+    @property
+    def temperature(self) -> int:
+        """Returns the temperature in degree Celsius as determined by the NTC thermistor.
+
+        Returns:
+            int: degree Celsius.
+        """
+
+        return self.data["wash_temp"]
+
+    @property
+    def power(self) -> bool:
+        """Returns the power status of the device.
+
+        Returns:
+            bool: The return value. True for power on, False for power off.
+
+        Raises:
+            DeviceException: When the returned value by the device is not 1 or 0.
+        """
+
+        value = self.data["power"]
+        if value in [0, 1]:
+            return bool(value)
+
+        raise DeviceException(f"{value} is not a valid power status.")
+
+    @property
+    def time_left(self) -> timedelta:
+        """Returns the timedelta in seconds of time left of the current program.
+
+        Will always be 0 if no program is running.
```

Comment:
Maybe check for this case in the code and return None, and convert the type hint to `Optional[timedelta]`?